File: LudoGame/LudoGame/Models/PlayerColor.swift
import SwiftUI

enum PlayerColor: String, CaseIterable, Identifiable, Codable {
    case red, green, yellow, blue

    var id: String { rawValue }

    var displayName: String {
        rawValue.capitalized
    }

    var color: Color {
        switch self {
        case .red: return Color(red: 0.86, green: 0.15, blue: 0.15)
        case .green: return Color(red: 0.13, green: 0.70, blue: 0.32)
        case .yellow: return Color(red: 0.98, green: 0.78, blue: 0.08)
        case .blue: return Color(red: 0.12, green: 0.47, blue: 0.95)
        }
    }

    var lightColor: Color {
        color.opacity(0.35)
    }

    /// Entry cell on the shared outer track (clockwise from red).
    var startPosition: Int {
        switch self {
        case .red: return 0
        case .green: return 13
        case .yellow: return 26
        case .blue: return 39
        }
    }

    /// Board corner used for yard placement and home column direction.
    var homeCorner: BoardCorner {
        switch self {
        case .red: return .bottomLeft
        case .green: return .topLeft
        case .yellow: return .topRight
        case .blue: return .bottomRight
        }
    }

    static let safePositions: Set<Int> = [0, 8, 13, 21, 26, 34, 39, 47]
}

enum BoardCorner {
    case topLeft, topRight, bottomLeft, bottomRight
}

File: LudoGameAndroid/app/src/main/java/com/ludogame/app/models/PlayerColor.kt
package com.ludogame.app.models

import androidx.compose.ui.graphics.Color

enum class BoardCorner {
    TOP_LEFT,
    TOP_RIGHT,
    BOTTOM_LEFT,
    BOTTOM_RIGHT
}

enum class PlayerColor(val displayName: String) {
    RED("Red"),
    GREEN("Green"),
    YELLOW("Yellow"),
    BLUE("Blue");

    val composeColor: Color
        get() = when (this) {
            RED -> Color(0xFFDB2626)
            GREEN -> Color(0xFF21B352)
            YELLOW -> Color(0xFFFAC714)
            BLUE -> Color(0xFF1F78F2)
        }

    val lightColor: Color
        get() = composeColor.copy(alpha = 0.35f)

    val startPosition: Int
        get() = when (this) {
            RED -> 0
            GREEN -> 13
            YELLOW -> 26
            BLUE -> 39
        }

    val homeCorner: BoardCorner
        get() = when (this) {
            RED -> BoardCorner.BOTTOM_LEFT
            GREEN -> BoardCorner.TOP_LEFT
            YELLOW -> BoardCorner.TOP_RIGHT
            BLUE -> BoardCorner.BOTTOM_RIGHT
        }

    companion object {
        val allCases = entries
        val safePositions = setOf(0, 8, 13, 21, 26, 34, 39, 47)
    }
}
